### PR TITLE
Fix field Values search for PKs

### LINKED ISSUE #32985: e.some is not a function breaks the FE on the filter modal
### Describe the bug

1) New question->Go to hosting insights->select the "pulses" table->filter modal
2) instance dns -> search for one, type a few chars --> hit the error

same as always
image

### To Reproduce

Above

### Expected behavior

We shouldn't break the FE

### Logs

_No response_

### Information about your Metabase installation

```JSON
master
```

### Severity

P1

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by qnkhuat**
Fix #32985 (see this [comment](https://github.com/metabase/metabase/issues/32985#issuecomment-1669404944) for how to reproduce)

See the comments for explaination

### CODE REVIEW

File: src/metabase/api/field.clj [hunk 1]
```diff
@@ -419,13 +419,14 @@
...
    (try
-     (let [field   (follow-fks field)
-           limit   (or maybe-limit default-max-field-search-limit)
-           results (qp/process-query (search-values-query field search-field value limit))]
-       (get-in results [:data :rows]))
-     (catch Throwable e
-       (log/error e (trs "Error searching field values"))
-       nil))))
+    (let [field        (follow-fks field)
+          search-field (follow-fks search-field)
```
Comment (qnkhuat) [inline src/metabase/api/field.clj:423]: the fix is just wrapping `search-field` with `follow-fks`

Comment (qnkhuat) [reply src/metabase/api/field.clj:423]: I know this is confused, and I don't think I explained the above problem well, feel free to ping me, I'll give you a walk through. Thank you!

**Review by calherries — APPROVED** [review]
Looks solid

File: src/metabase/api/field.clj [hunk 2]
```diff
@@ -419,13 +419,14 @@
...
+          limit        (or maybe-limit default-max-field-search-limit)
+          results      (qp/process-query (search-values-query field search-field value limit))]
```
[the remark this answers is not in the document]
Comment (qnkhuat) [reply src/metabase/api/field.clj:427]: I added a note in  `search-values-query`'s docstring here https://github.com/metabase/metabase/pull/33006/commits/3880116443e87197bc166b83fc7b09c79c7600d9

**Pull request merged**